testsrc/main/mondrian/olap/HierarchyBugTest.java

```diff
@@ -72,6 +77,258 @@ public void testNoHierarchy() {
+        org.olap4j.metadata.Member year1997 =
+            positions.get(0).getMembers().get(0);
+        String year1997HierarchyName = year1997.getHierarchy().getUniqueName();
+        assertEquals("[Time].[Weekly]", year1997HierarchyName);
+
+        org.olap4j.metadata.Level year = year1997.getLevel();
+        String yearHierarchyName = year.getHierarchy().getUniqueName();
+        assertEquals(year1997HierarchyName, yearHierarchyName);
+        flushContextSchemaCache();
+    }
+
+    public void testNameSsasInCompatibleOlap4j() throws SQLException {
+        // SsasCompatibleNaming defaults to false
+        String mdx = "SELECT\n"
+            + "   [Measures].[Unit Sales] ON COLUMNS,\n"
+            + "   [Time].[Time].[Year].Members ON ROWS\n"
+            + "FROM [Sales]";
+        CellSet result = getTestContext().executeOlap4jQuery(mdx);
+
+        List<org.olap4j.Position> positions =
+            result.getAxes().get(1).getPositions();
+        org.olap4j.metadata.Member year1997 =
+            positions.get(0).getMembers().get(0);
+        String year1997HierarchyName = year1997.getHierarchy().getUniqueName();
+        assertEquals("[Time]", year1997HierarchyName);
+
+        org.olap4j.metadata.Level year = year1997.getLevel();
+        String yearHierarchyName = year.getHierarchy().getUniqueName();
+        assertEquals(year1997HierarchyName, yearHierarchyName);
+        flushContextSchemaCache();
+    }
+
+    public void testNameSsasInCompatibleOlap4jNamedHierarchy()
+        throws SQLException
+    {
+        // SsasCompatibleNaming defaults to false
+        String mdx =
+            "SELECT\n"
+            + "   [Measures].[Unit Sales] ON COLUMNS,\n"
+            + "   [Time.Weekly].[Year].Members ON ROWS\n"
+            + "FROM [Sales]";
+        CellSet result = getTestContext().executeOlap4jQuery(mdx);
+
+        List<org.olap4j.Position> positions =
+            result.getAxes().get(1).getPositions();
+        org.olap4j.metadata.Member year1997 =
+            positions.get(0).getMembers().get(0);
+        String year1997HierarchyName = year1997.getHierarchy().getUniqueName();
+        assertEquals("[Time.Weekly]", year1997HierarchyName);
+
+        org.olap4j.metadata.Level year = year1997.getLevel();
+        String yearHierarchyName = year.getHierarchy().getUniqueName();
+        assertEquals(year1997HierarchyName, yearHierarchyName);
+        flushContextSchemaCache();
+    }
+
+    public void testNameSsasInCompatibleOlap4jDateDim()
+        throws SQLException
+    {
+        // SsasCompatibleNaming defaults to false
+        String mdx =
+            "SELECT\n"
+            + "   [Measures].[Unit Sales] ON COLUMNS,\n"
+            + "   [Time.Weekly].[Year].Members ON ROWS\n"
+            + "FROM [Sales]";
+        TestContext context = getTestContext();
+        String dateDim  =
+            "<Dimension name='Date' type='TimeDimension'>\n"
+            + "    <Hierarchy hasAll='false' primaryKey='time_id'>\n"
```

Comment:
@YuryBY 
You've added a [Date] dimension, but your query and test is still validating against the [Time] dimension.  I'm guessing you intended to use the [Date] dim.

One other thing-- all these tests have a whole lot of overlap.  Could you extract out some of the commonality so the tests are a bit more succinct?